src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,130 @@
 package core.basesyntax;
 
+import java.util.Map;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private int threshold;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> newNode;
+        if (key == null) {
+            newNode = new Node<>(0, key, value, null);
+        } else {
+            newNode = new Node<>(key.hashCode(), key, value, null);
+        }
+        int position = hash(key);
+        if (table[position] == null) {
+            table[position] = newNode;
+        } else if (Objects.equals(table[position].key, key)) {
+            table[position].value = newNode.value;
+            return;
+        } else {
+            Node<K, V> itemInTable = table[position];
+            while (itemInTable.next != null) {
+                itemInTable = itemInTable.next;
+                if (Objects.equals(itemInTable.key, key)) {
+                    itemInTable.value = newNode.value;
+                    return;
+                }
+            }
+            itemInTable.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int position = hash(key);
+        if (table[position] == null) {
+            return null;
+        }
+        if (Objects.equals(table[position].key, key)) {
+            return table[position].value;
+        } else {
+            Node<K, V> itemInTable = table[position];
+            while (itemInTable.next != null) {
+                itemInTable = itemInTable.next;
+                if (Objects.equals(itemInTable.key, key)) {
+                    return itemInTable.value;
+                }
+            }
+        }
```

Comment:
if you write the check `itemInTable != null` in the loop, then you will not need the if else construction